inc/thorvg.h

```diff
@@ -152,6 +153,8 @@ class TVG_EXPORT Canvas
     virtual Result sync() noexcept;
 
     _TVG_DECLARE_PRIVATE(Canvas);
+
+    friend class SwCanvas;
```

Comment:
SwCanvas is a derived, base class should not depend on it...

Reply:
without friendship I'm not able to switch flag in target API in SwClass. 


```c
Result SwCanvas::target(uint32_t* buffer, uint32_t stride, uint32_t w, uint32_t) {
//...
Canvas::pImpl->forceUpdateAll = true;
}
```

Reply:
If you don't like this solution, I can add virtual Canvas::target() method, but in this case it would be great to change also tvg_swcanvas_create() -> tvg_canvas_create() and tvg_swcanvas_set_target() to tvg_canvas_set_target() (this solution breaks backward compatibility). 

Maybe do you know better option ? I can consider other possibilities if you want